src/dashboard-refactor/lists-sidebar/components/lists-search-bar/index.tsx

```diff
@@ -0,0 +1,173 @@
+
+import styles, { fonts } from 'src/dashboard-refactor/styles'
+import colors from 'src/dashboard-refactor/colors'
+import { SidebarLockedState } from '../../types'
+
+const textStyles = `
+    font-family: ${fonts.primary.name};
+    font-weight: ${fonts.primary.weight.normal};
+    font-size: 10px;
+    line-height: 15px;
+    color: ${fonts.primary.colors.primary};
+    cursor: text;
+`
+
+const OuterContainer = styled.div<{ isSidebarLocked: boolean }>`
+    height: min-content;
+    padding-left: 8px;
+    padding-right: 8px;
+    ${(props) =>
+        css`
+            background-color: ${props.isSidebarLocked
+                ? colors.white
+                : colors.lightMidGrey};
+        `}
+    border-radius: 3px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+`
+
+const InnerContainer = styled.div<{ displayTopBorder?: boolean }>`
+    height: 24px;
+    width: 100%;
+    background-color: transparent;
+    display: flex;
+    flex-direction: row;
+    justify-content: start;
+    align-items: center;
+    ${(props) =>
+        props.displayTopBorder &&
+        css`
+            border-top: 0.5px solid ${colors.lighterGrey};
+        `}
+`
+
+const Input = styled.input<{ isFocused: boolean }>`
+    ${textStyles}
+    width: 100%;
+    border: none;
+    background: inherit;
+    ${(props) => {
+        const { primary, secondary } = fonts.primary.colors
+        return css`&::placeholder {
+            ${textStyles}
+            color: ${props.isFocused ? secondary : primary};
+            `
+    }}}
+    &:focus {
+        outline: none;
+    }
+`
+
+const TextSpan = styled.span<{ bold?: boolean }>`
+    ${textStyles}
+    ${(props) =>
+        props.bold &&
+        css`
+            font-weight: ${styles.fonts.primary.weight.bold};
+        `}
+`
+
+const IconContainer = styled.div`
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-items: start;
+`
+
+const StyledIcon = styled(Icon)`
+    color: #3a2f45;
+`
+export interface ListsSidebarSearchBarProps {
+    isSearchBarFocused: boolean
+    hasPerfectMatch: boolean
+    searchQuery?: string
+    onFocus(): void
+    onSearchQueryChange(inputString: string): void
+    onInputClear(): void
+    onCreateNew(newListName: string): void // should this return a promise?
+    sidebarLockedState: SidebarLockedState
+}
+
+export default class ListsSidebarSearchBar extends PureComponent<
+    ListsSidebarSearchBarProps
+> {
+    inputRef = React.createRef<HTMLInputElement>()
+    componentDidMount = () => {
+        if (this.props.isSearchBarFocused) this.inputRef.current.focus()
+    }
+    handleInputChange: React.ChangeEventHandler = (
+        evt: React.ChangeEvent<HTMLInputElement>,
+    ) => {
+        this.props.onSearchQueryChange(evt.currentTarget.value)
+    }
+    handleCreateNewClick: React.MouseEventHandler = (evt: React.MouseEvent) => {
```

Comment:
Declaring this function property as `React.MouseEventHandler` type will automatically type the event as `React.MouseEvent`. So you don't need to also manually type the event here. Same with the `handleInputChange` property. `render` method doesn't need to declare that it will return `JSX.Element` either as the class it is in extends `React.PureComponent` so TS will automatically work out this type.

If you use VSCode, you can cmd/ctrl click these types to check out how they're actually defined and see what other goodies will be automatically inferred from using them.

Reply:
I think I could use a bit of a primer on React + Typescript. I hadn't used Typescript before working on Memex and have just picked it up as I go along. Are there any resources you'd recommend?